packages/beacon-node/src/network/processor/extractSlotRootFns.ts

```diff
@@ -0,0 +1,35 @@
+import {SlotRootHex} from "@lodestar/types";
+import {
+  getBlockRootFromAttestationSerialized,
+  getBlockRootFromSignedAggregateAndProofSerialized,
+  getSlotFromAttestationSerialized,
+  getSlotFromSignedAggregateAndProofSerialized,
+} from "../../util/sszBytes.js";
+import {nullableResult} from "../../util/wrapError.js";
+import {GossipType} from "../gossip/index.js";
+import {ExtractSlotRootFns} from "./types.js";
+
+/**
+ * Extract the slot and block root of a gossip message form serialized data.
+ * Only applicable for beacon_attestation and beacon_aggregate_and_proof topics.
+ */
+export function createExtractBlockSlotRootFns(): ExtractSlotRootFns {
+  return {
+    [GossipType.beacon_attestation]: nullableResult(getSlotRootFromAttestationSerialized),
```

Comment:
Let's define the API directly as return base64 string or null if invalid, no need to wrap with this function if there's a single error case